src/core/meshcop/tcat_agent.cpp

```diff
@@ -500,6 +503,66 @@ Error TcatAgent::HandleStartThreadInterface(void)
     return error;
 }
 
+uint8_t *TcatAgent::GetAdvertisementData(uint16_t &aLen)
+{
+    Message *buffer = Get<MessagePool>().Allocate(Message::kTypeBle, 0);
+
+    if (buffer == nullptr || mVendorInfo == nullptr)
+    {
+        aLen = 0;
+        return nullptr;
+    }
+
+    *reinterpret_cast<uint16_t *>(mAdvertisement) = LittleEndian::HostSwap16(OT_TOBLE_SERVICE_UUID);
+    mAdvertisement[2]                             = OT_TCAT_VERSION << 4 | OT_TCAT_OPCODE;
+
+    IgnoreError(buffer->AppendBytes(mAdvertisement, 3));
+
+    switch (MapEnum(mVendorInfo->mDeviceIdType))
+    {
+    case kTcatDeviceIdOui24:
+        IgnoreError(ot::Tlv::Append<VendorOuiTlv24>(*buffer, mVendorInfo->mDeviceId));
+        break;
+    case kTcatDeviceIdOui36:
+        IgnoreError(ot::Tlv::Append<VendorOuiTlv36>(*buffer, mVendorInfo->mDeviceId));
+        break;
+    case kTcatDeviceIdDiscriminator:
+        IgnoreError(ot::Tlv::Append<DeviceDiscriminatorTlv>(*buffer, mVendorInfo->mDeviceId));
+        break;
+    case kTcatDeviceIdIanaPen:
+        IgnoreError(ot::Tlv::Append<VendorIanaPenTlv>(*buffer, mVendorInfo->mDeviceId));
+        break;
+    default:
+        break;
+    }
+
+    otBleLinkCapabilities caps = otPlatGetBleLinkCapabilities(&GetInstance());
+
+    IgnoreError(ot::Tlv::Append<BleLinkCapabilitiesTlv>(*buffer, *reinterpret_cast<uint8_t *>(&caps)));
```

Comment:
This TLV should not be included if all bits are zero. (Just for efficiency)

Reply:
Just for clarification, the S bit should be based on what value or property?

Reply:
> This TLV should not be included if all bits are zero. (Just for efficiency)

Do we want this behavior for sure? This structure contains the Thread network active bit and rxOnWhenIdle. This means that TLV will not be included only not active SED or SSED devices.

Reply:
@canisLupus1313 It seems there was some mix-up here of TLVs in the comments. My comment was only for the BLE Link Capabilities TLV that contains only 1 bit currently. (G bit)  The spec says that it's only included if 1 or more bits are '1'.  Arnulf's comment was for another TLV, the Thread Device Type and Status TLV.

Reply:
@canisLupus1313 @arnulfrupp  About the question how to determine the S bit value -- I checked the public functions available in `dataset_manager.hpp` (see below). I don't fully understand what these APIs provide and how it relates to what the Thread spec defines. They seem to be different in criteria.  Strongly suggest to use a separate PR to add the S bit !! (This is complex and may need additional OT properties.)

```
/**
     * Indicates whether the Active Dataset is partially complete.
     *
     * Is primarily used to determine whether a user has supplied a partial Active Dataset for use
     * with joining a network.
     *
     * @retval TRUE   If an Active Dataset is saved but does not include an Active Timestamp.
     * @retval FALSE  If an Active Dataset is not saved or does include an Active Timestamp.
     *
     */
    bool IsPartiallyComplete(void) const;

    /**
     * Indicates whether the Active Dataset is complete.
     *
     * @retval TRUE   If an Active Dataset is saved and includes an Active Timestamp.
     * @retval FALSE  If an Active Dataset is not saved or does include an Active Timestamp.
     *
     */
    bool IsComplete(void) const;

    /**
     * Indicates whether or not a valid network is present in the Active Operational Dataset.
     *
     * @retval TRUE if a valid network is present in the Active Dataset.
     * @retval FALSE if a valid network is not present in the Active Dataset.
     *
     */
    bool IsCommissioned(void) const;
```

Reply:
Noted BLE link capabilities were adjusted.
S bit is only included in the `DeviceTypeAndStatus` for completeness of structure.